syntax = "proto2";
package proto.landlord;
import "game.proto";

//开始抢地主
message RobLandlordInfo
{
  required int32 operatorid = 1;        //当前操作id
  repeated int32 can_rod_scores = 2;    //可以抢的分数
  repeated int32 rod_scores = 3;        //所有玩家抢的分数
  optional bool isfinish = 4;           //抢地主是否完成
}

//开始抢地主
message RobLandlord
{
  required int32 seatid = 1;                        //座位id
  required int32 rod_score = 2;                     //当前玩家抢的分数
  optional RobLandlordInfo  roblandlord_info = 3;   //抢庄信息
}

//广播抢地主结果
message AckRobLandlordResult
{
  required int32 seatid = 1;         //座位id
  required int32 rod_value = 2;      //最高赔率
  repeated int32 public_cards = 3;   //底牌
}

//广播抢地主结果
message LandlordScene
{
  required game.GameScene public_scene = 1;       //公共游戏场景
  repeated game.HandCards hand_cards = 2;         //玩家信息
  optional int32 max_rod_score = 3;               //最高分数
  repeated int32 public_cards = 4;                //底牌
  optional int32 roblandlord_state = 5;           //抢地主状态                                              //抢地主状态
  optional RobLandlordInfo roblandlord_info = 6;  //抢地主分数
  optional int32 operator_id = 7;                 //当前操作id
}

//广播抢地主结果
message AckOutCard
{
  required int32 out_seatid = 1;              //出牌id
  required int32 operator_seatid = 2;         //当前操作id
  repeated int32 out_cards = 3;               //玩家出牌
  optional int32 isfirst = 4;                 //第一手或最大牌
}

//广播抢地主结果
message ReqOutCard
{
  repeated int32 out_cards = 1;               //玩家出牌
}